sms_alternative_provider/models/sms_api.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2024 Hunki Enterprises BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3.0)
+
+from odoo import api, models
+
+
+class SmsApi(models.AbstractModel):
+    _inherit = "sms.api"
+
+    @api.model
+    def _send_sms_batch(self, messages):
+        """
+        Divert SMS sending to the gateway model if not asked not to
+        """
+        if self.env.context.get("force_iap"):
+            return super()._send_sms_batch(messages)
+        return self.env["ir.sms.gateway"]._send(
+            [dict(message, id=message.get("res_id")) for message in messages],
+            handle_results=False,
```

Comment:
So in some cases, handle_results=False, which does not call handle_results directly after sending; instead it's being called from postprocess_iap_results, but then without the original message dicts.

What are the scenarios for having handle_results=True or False?

Reply:
we want to support just saying

```
self.env["ir.sms.gateway"]._send([some dicts, may or may not link to sms.sms records])
```

in which case the send function should call `_handle_results`, and here it shouldn't because `_postprocess_iap_sent_sms` does it. So this clumsiness is caused by this module trying to introduce a new kind of sms api, while staying compatible with overrides other modules might do for the original api. We could consider dropping this.